### PR TITLE
changes to assessment

### PR DESCRIPTION
**Pull request by Chrisluna10**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -14,21 +14,50 @@ var temp1 = 35
...
+//     return `${degree} is below boiling point`
+// }
+// if degree is equal to 213, return "(degree) is at boiling point."
+//     else {
+//     return `${degree} is at boiling point`
+// } 
+// }
+// console.log(temp(221))
+// console.log(temp(212))
+// console.log(temp(9))
```
Comment (WalkerTexas-coder) [inline code-challenges.js:34]: Good Work!

File: code-challenges.js [hunk 2]
```diff
@@ -14,21 +14,50 @@ var temp1 = 35
...
+
 
 
 // --------------------3) Create a function that takes in a string of a single word and returns the string with all letters reversed.
 // Use the test variables provided below. Expected output: "ovarb", "eilrahc"
 
 var myString1 = "bravo"
 var myString2 = "charlie"
+// combine .split to break up each letter into its own array then reverse the array with .reverse and finally join each letter back together to return the required output. 
+// I tried creating a function around this combination but I was unable to.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:58]: What happened when you did? the logic looks good!

File: code-challenges.js [hunk 3]
```diff
@@ -37,18 +66,36 @@ var myString2 = "charlie"
...
-var numberExample3 = 7
-var numberExample4 = 19
-
+// var numberExample3 = 7
+// var numberExample4 = 19
 
+// const subtraction = (num1, num2) =>{
+//     return num1 - num2
+// }
+// console.log(subtraction(42, 27))
```
Comment (WalkerTexas-coder) [inline code-challenges.js:82]: What happens when we subrtact numberExample3 from numberExample4? Do we get the expected output?

File: code-challenges.js [hunk 4]
```diff
@@ -37,18 +66,36 @@ var myString2 = "charlie"
...
 // --------------------5) Copy/paste your code from #4. Refactor your code to also accept non-number edge cases. Inform your user if the variable passed into the subtraction function is not a number.
 // Use the two sets of test variables provided below. Expected output: "Your input is not a number", 3
 // Hint: There is an operator in JavaScript called typeof :)
 
 var numberExampleRefactor1 = 42
 var numberExampleRefactor2 = "hello"
-
-var numberExampleRefactor3 = 27
-var numberExampleRefactor4 = 24
+// tried including type of in different areas but I was not able to get it to work.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:90]: What did that look like?

File: coding-concepts.js [hunk 5]
```diff
@@ -29,23 +29,23 @@ var languages = ["JavaScript", "Ruby", "Python", "C++"]
...
 
 // --------------------4) What will this log?
 
 var weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Saturday, Sunday
+// b) Verify and explain: There was an error because you can not use the uppercase function on a variable.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:42]: We can use it on a variable but not on certain kinds. Whats happening here?

File: interview-questions.md [hunk 6]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a statement that performs a task. After the task is performed an output is recieved. This would be used when a specific code is needed to be executed multiple times.
 
-  Researched answer:
+  Researched answer: Functions are a reusable piece of code that always return an output only when executed. Functions are used to define a block of code and execute as many times as you want.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:12]: correct, they can encapsulate logic so that we can consistently manipulte data.

File: interview-questions.md [hunk 7]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Functions are a reusable piece of code that always return an output only when executed. Functions are used to define a block of code and execute as many times as you want. 
 
 
 
 2. In regards to functions, what is the difference between an argument and a parameter?
 
-  Your answer:
+  Your answer: A parameter is a placeholder inside of a function while an argument is the actual value inputed into the function.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:18]: Great!

File: interview-questions.md [hunk 8]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: To keep us from reinventing the wheel, the JavaScript language has a bunch of pre-made functionality that can be used on the data type string. These are called built-in methods. The built-in methods often rely on properties that are innate to the data type of string.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Proper pair programming requires only one person to be coding at a time while the other person provides information such as questions, corrections and code to input while watching the other person code. The two programmers are also switching frequently in order to have adequete breaks from one role. The person that is coding is called the driver while the person observing and managing is called the navigator.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:42]: Awesome!

**Review by WalkerTexas-coder — COMMENTED** [review]
Chris, good work on this assessment. I can see you're handleing a lot of new information really well. Your logic is super solid and it's cool to follow along with your solutions. I would like to see more psuedo coding next week. Communicating your problem solving process is not just a benefit to other developers but also yourself down the line. With that I have left a few comments with some questions I would apperciate your response too. No need to update the code just reply in thread and I will approve the PR.

File: code-challenges.js (hunk 2 above)
Comment (Chrisluna10) [reply code-challenges.js:58]: I believe I fixed it after I commented that so it should be good lol.

File: code-challenges.js (hunk 3 above)
Comment (Chrisluna10) [reply code-challenges.js:82]: the output is -12. This is not correct because I did not make a statement that would reverse the function if num3 was smaller than num4

File: code-challenges.js (hunk 4 above)
Comment (Chrisluna10) [reply code-challenges.js:90]: The output returned NaN because I did not format my function correctly.

File: coding-concepts.js (hunk 5 above)
Comment (Chrisluna10) [reply coding-concepts.js:42]: This is an array so toUpperCase did not work.
